[logs] Add dynamo_timed to get better compilation time breakdown for AOTI

henrylhtsang opened the pull request:
Adding some dynamo timed for the purpose of better understanding AOTI compilation time.

Probably would require a few more passes. A lot of time is spent in Scheduler.__init__, and not enough annotations are there.

run_command_and_check takes a lot time as well. But there is probably not much we can do. Maybe we can add a config to tune C++ optimization level?

traces:
<img width="1205" alt="Screenshot 2024-11-08 at 4 41 10 PM" src="https://github.com/user-attachments/assets/61645264-b3af-4d4a-804d-700b0f831c7c">


Differential Revision: D65554141


cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang @aakhundov

desertfire commented on torch/_inductor/codecache.py:
```diff
@@ -2006,7 +2007,8 @@ def _pad_to_alignment(raw_bytes: bytes) -> bytes:
                     os.chmod(output_o, 0o644)
                 else:
-                    run_command_and_check(compile_cmd)
+                    with dynamo_timed("run_compile_cmd", log_pt2_compile_event=True):
```
Didn't you already add dynamo_timed to run_command_and_check?

henrylhtsang replied:
I think I added them so I can annotate which run_command_and_check it is. But let me think about it more next week. 

<img width="1235" alt="Screenshot 2024-11-12 at 2 44 44 PM" src="https://github.com/user-attachments/assets/648b766f-4949-4a36-b166-b43285b6d28f">

desertfire commented:
LGTM, but I will let folks who work on compilation time to comment, @masnesral , @jamesjwu

henrylhtsang commented:
@desertfire removed the extra dynamo_tiled outside of run_command_and_check

masnesral commented:
> LGTM, but I will let folks who work on compilation time to comment, @masnesral , @jamesjwu

Looks fine to me too; the only thing I don't know is if we want all of these in pt2 compile events? @jamesjwu, yes?

henrylhtsang committed:
[logs] Add logs for AOTI compilation time (#140198)

Summary:



Below line forces Sandcastle to run only specified contbuilds.
build_only[github-export-checks,executorch,pytorch_benchmark,pytorch_benchmark_fb,pytorch_quantization,pytorch_distributed,pytorch_distributed_gpu,pytorch_dynamo,pytorch_inductor,pytorch_inductor_fb,pytorch_functorch,pytorch_fx2trt,pytorch_diff_train_tests_ads,glow_fb_pytorch_tests,training_platform,training_platform_compatibility,training_toolkit_applications,training_toolkit_examples,training_toolkit_model_optimization,dper3_pytorch,xplat_caffe2,pytorch_dev,android-pytorch-instrumentation-tests,smartpytorchgithub_first_try_merge,frl-target-determinator,f6-buck,training_platform_for_github,sigmoid_cpu,sigmoid_gpu,aiplatform_modelprocessing_for_github,accelerators_workloads_models_slimdsnn,ae_aotinductor_benchmark_test,aps_,apf,aps_deterministic_ne_tests,dper_lib_silvertorch,torchrec,torchrec_fb,deeplearning_aot_inductor,aiplatform_modelstore]
#skipfbcodelongtail
pytorch-oss-diff-train

Differential Revision: D65554141

henrylhtsang commented:
@pytorchbot merge
